### PR TITLE
Improved protocol version support for VS project

### PR DESCRIPTION
**Pull request by dmkozh**
# Description

- Introduced build configurations for current and next protocol versions
- Added both current and next version XDR/generated headers to the project
- Use the appropriate XDRs for appropriate protocol versions
- Libraries only have one Debug/Release configuration so that they don't need to be rebuilt

# Checklist
- [ ] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [ ] Rebased on top of master (no merge commits)
- [ ] Ran `clang-format` v8.0.0 (via `make format` or the Visual Studio extension)
- [ ] Compiles
- [ ] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

File: Builds/VisualStudio/stellar-core.vcxproj [hunk 1]
```diff
@@ -127,6 +166,65 @@
...
+    <PreBuildEvent>
+      <Command>
+      </Command>
+    </PreBuildEvent>
+    <PreBuildEvent>
+      <Message>
+      </Message>
+    </PreBuildEvent>
+    <CustomBuildStep>
+      <Inputs>src/generated/xdr/Stellar-overlay.h</Inputs>
```
Comment (MonsieurNicolas) [inline Builds/VisualStudio/stellar-core.vcxproj:207]: I think this is incorrect (this is the kind of bug that was fixed in master recently)

File: Builds/VisualStudio/stellar-core.sln [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio Version 16
-VisualStudioVersion = 16.0.28803.202
+# Visual Studio Version 17
+VisualStudioVersion = 17.1.32421.90
```
Comment (MonsieurNicolas) [inline Builds/VisualStudio/stellar-core.sln:3]: why did this change? (so far I don't think changes caused this bump)

File: Builds/VisualStudio/stellar-core.vcxproj (hunk 1 above)
Comment (MonsieurNicolas) [reply Builds/VisualStudio/stellar-core.vcxproj:207]: should grep for `src/generated/xdr`

Comment (MonsieurNicolas) [reply Builds/VisualStudio/stellar-core.vcxproj:207]: should we just make all those rules the same and use a variable? basically PR becomes:
replace `src/generated` by `src/$(Configuration)/generated`  which gets rid of the rust "generated" folder as well.
And this I think can be done across the board.

**Review by MonsieurNicolas — CHANGES_REQUESTED** [review]

File: Builds/VisualStudio/stellar-core.vcxproj (hunk 1 above)
Comment (dmkozh) [reply Builds/VisualStudio/stellar-core.vcxproj]: Changed everything to use $(Configuration). The only downside is that VS index/navigation isn't updated properly (more details in commit message), but that issue must have existed for the previous approach too.

File: Builds/VisualStudio/stellar-core.sln (hunk 2 above)
Comment (dmkozh) [reply Builds/VisualStudio/stellar-core.sln]: Must be some automatic VS stamp, reverted.

**Comment by MonsieurNicolas** [comment]
Looks good (and tried it locally) but it needs to be rebased

**Comment by MonsieurNicolas** [comment]
r+ 56c7292278257057d0d7e010368d56cf5ba4e241

**Pull request merged**